server/protocols/webadmin-cli/src/main/java/org/apache/james/cli/user/UserExistCommand.java

```diff
@@ -0,0 +1,68 @@
+/******************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one     *
+ * or more contributor license agreements.  See the NOTICE file   *
+ * distributed with this work for additional information          *
+ * regarding copyright ownership.  The ASF licenses this file     *
+ * to you under the Apache License, Version 2.0 (the              *
+ * "License"); you may not use this file except in compliance     *
+ * with the License.  You may obtain a copy of the License at     *
+ *                                                                *
+ * http://www.apache.org/licenses/LICENSE-2.0                     *
+ *                                                                *
+ * Unless required by applicable law or agreed to in writing,     *
+ * software distributed under the License is distributed on an    *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY         *
+ * KIND, either express or implied.  See the License for the      *
+ * specific language governing permissions and limitations        *
+ * under the License.                                             *
+ ******************************************************************/
+
+package org.apache.james.cli.user;
+
+import java.util.concurrent.Callable;
+
+import org.apache.james.cli.WebAdminCli;
+import org.apache.james.httpclient.UserClient;
+
+import feign.Feign;
+import feign.Response;
+import picocli.CommandLine;
+
+@CommandLine.Command(
+    name = "exist",
+    description = "Check if a user exists")
+public class UserExistCommand implements Callable<Integer> {
+
+    public static final int EXISTED_CODE = 200;
+    public static final int USER_NAME_INVALID_CODE = 400;
+    public static final int NOT_EXISTED_CODE = 404;
+
+    @CommandLine.ParentCommand UserCommand userCommand;
+
+    @CommandLine.Parameters String userName;
+
+    @Override
+    public Integer call() {
+        try {
+            UserClient userClient = Feign.builder()
+                .target(UserClient.class, userCommand.webAdminCli.jamesUrl + "/users");
+            Response rs = userClient.doesExist(userName);
+            if (rs.status() == EXISTED_CODE) {
+                userCommand.out.println(userName + " exists");
+                return WebAdminCli.CLI_FINISHED_SUCCEED;
+            } else if (rs.status() == USER_NAME_INVALID_CODE) {
+                userCommand.out.println("The user name is invalid");
+                return WebAdminCli.CLI_FINISHED_SUCCEED;
```

Comment:
Shouldn't that be a failure then instead of a success?

Reply:
I think in this case: user exist command need to return true or false(does a username exist). although it returns false, the CLI still finished its job, so it succeed